### ISSUE TITLE
AsOf schema names

### ISSUE THREAD

**Issue by fnatte** [anymaniax/orval#245]
## What are the steps to reproduce this issue?

Generate client using spec:
```yaml
openapi: '3.0.0'
info:
  version: 1.0.0
  title: Test
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /colors:
    get:
      summary: List colors
      operationId: listColors
      tags:
        - colors
      responses:
        '200':
          description: An array of colors
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Color'
components:
  schemas:
    ColorType:
      type: string
      enum:
        - red
        - green
        - blue
    Color:
      type: object
      required:
        - color
      properties:
        type:
          allOf:
            - $ref: '#/components/schemas/ColorType'
```

## What happens?

The result is:

```typescript
export type ColorType = 'red' | 'green' | 'blue';


export const ColorType = {
  red: 'red' as ColorType,
  green: 'green' as ColorType,
  blue: 'blue' as ColorType,
};

export type ColorType = ColorType

export const ColorType = {...ColorType};

export interface Color {
  type?: ColorType;
}
```

The second type-const pair uses the same name as the first pair, resulting in `Cannot redeclare block-scoped variable 'ColorType'.ts (2451)`

## What were you expecting to happen?

Some other name of the second type. Maybe keep `ColorTypeAllOf`?

## Any other comments?

I traced the issue to here: https://github.com/anymaniax/orval/blob/master/src/core/resolvers/value.ts#L22
And noticed that the `name` parameter to `resolveValue` is `ColorTypeAllOf`, but the returned `name` from `resolveRef` is `ColorType`.

## What versions are you using?
49290694a8e072ee4f7311c0a04ee92da9cfef35

**Comment by dennisroche** [comment]
@anymaniax is this up-for-grabs?

**Comment by dennisroche** [comment]
2 x PRs. 

- #248 with the actual fix; and 
- #249 with some changes made to enable debugging of the tests

**Comment by dennisroche** [comment]
sorry, i'm going to have leave this bug for someone else to work on.

**Comment by anymaniax** [comment]
@dennisroche no worries I will do it when I have a bit of time 😀 and also your problem with eslint maybe open an other issue for that?

**Comment by anymaniax** [comment]
Should be fixed with `6.4.0` can you test when you have time, please?

**Comment by anymaniax** [comment]
Closing this issue since it's normally fixed. Don't hesitate if you have any problem
